### ISSUE TITLE
1.54 bug Area buff effect

### ISSUE THREAD

**Issue by DimasjkTV** [suriyun-production/mmorpg-kit-docs#287]
If you deactivate this checkmark, then the character also does not receive a buff
![image](https://user-images.githubusercontent.com/47291794/77052394-d32f4200-69dd-11ea-8948-91fab717574d.png)

**Comment by insthync** [comment]
Fix it by add codes to BaseMapInfo.cs → IsAlly function line 30, So the codes will be
```
        public bool IsAlly(BaseCharacterEntity characterA, BaseCharacterEntity characterB)
        {
            if (characterA == characterB)
                return true;
            if (characterA is BasePlayerCharacterEntity)
                return IsPlayerAlly(characterA as BasePlayerCharacterEntity, characterB);
            if (characterA is BaseMonsterCharacterEntity)
                return IsMonsterAlly(characterA as BaseMonsterCharacterEntity, characterB);
            return false;
        }
```
And also add codes to IsEnemy function
```
        public bool IsEnemy(BaseCharacterEntity characterA, BaseCharacterEntity characterB)
        {
            if (characterA == characterB)
                return false;
            if (characterA is BasePlayerCharacterEntity)
                return IsPlayerEnemy(characterA as BasePlayerCharacterEntity, characterB);
            if (characterA is BaseMonsterCharacterEntity)
                return IsMonsterEnemy(characterA as BaseMonsterCharacterEntity, characterB);
            return false;
        }
```

**Comment by DimasjkTV** [comment]
Works. thanks!

**Comment by insthync** [comment]
Fixed in 1.54b
